recipes/libhal/all/conanfile.py

```diff
@@ -0,0 +1,73 @@
+from conan import ConanFile
+from conan.tools.files import get, copy
+from conan.tools.layout import basic_layout
+from conan.tools.build import check_min_cppstd
+from conan.errors import ConanInvalidConfiguration
+import os
+
+
+required_conan_version = ">=1.50.0"
+
+
+class LibHALConan(ConanFile):
+    name = "libhal"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://libhal.github.io/libhal"
+    description = ("A collection of interfaces and abstractions for embedded "
+                   "peripherals and devices using modern C++")
+    topics = ("peripherals", "hardware", "abstraction", "devices", "hal")
+    settings = "os", "compiler", "arch", "build_type"
+    no_copy_source = True
+
+    def package_id(self):
+        self.info.clear()
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    @property
+    def _min_cppstd(self):
+        return "20"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "11",
+            "Visual Studio": "17",
+            "msvc": "19.22",
+            "clang": "13",
+            "apple-clang": "13.1.6"
+        }
+
+    def validate(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+        def lazy_lt_semver(v1, v2):
+            lv1 = [int(v) for v in v1.split(".")]
+            lv2 = [int(v) for v in v2.split(".")]
+            min_length = min(len(lv1), len(lv2))
+            return lv1[:min_length] < lv2[:min_length]
+        compiler = str(self.settings.compiler)
+        version = str(self.settings.compiler.version)
+        minimum_version = self._compilers_minimum_version.get(compiler, False)
+        if not minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.name} requires C++20. Your compiler configuration ({compiler}-{version}) wasn't validated. \
+                please report an issue if it does actually supports c++20.")
+        elif lazy_lt_semver(version, minimum_version):
+            raise ConanInvalidConfiguration(
+                f"{self.name} {self.version} requires C++20, which your compiler ({compiler}-{version}) does not support")
+
+    def layout(self):
+        basic_layout(self)
+
```

Comment:
```suggestion

    def source(self):
        get(self, **self.conan_data["sources"][self.version],
            destination=self.source_folder, strip_root=True)

```

source() here to understand the flow of execution when reading the recipe